Documentation/coreos.rst

```diff
@@ -0,0 +1,376 @@
+
+
+Step 2: Setup Kubelet
+--------------------------------
+
+On CoreOS, Kubelet (the Kubernetes agent that runs on each node, more info `here <https://kubernetes.io/docs/admin/kubelet/>`_) runs as a container. In order to securely communicate with the API server, kubelet uses the TLS assets we generated as part of Step 1.
+
+Master nodes are usually not scheduled to run workloads, so we provide the ``-register-schedulable=false`` in the example YAML excerpt below. This ensures that workloads are only scheduled on worker nodes.
+
+Since we are setting up Kubelet to use Cilium, we want to configure its networking to utilize CNI (Container Networking Interface). This ensures that each pod that is created can communicate with one another within the cluster with Cilium networking configured.
+
+* Replace ${ADVERTISE_IP} with this node's publicly routable IP.
+* Replace ${DNS_SERVICE_IP}
+* Replace ${KUBE_VERSION} with a version  ``>= 1.6.4``.
+* Cilium-specific configuration    
+
+    * Mount the CNI configuration directory you created in step 1 so Kubelet can pick up the CNI configuration from the host filesystem:
+
+         ::
+
+             --volume etc-cni,kind=host,source=/etc/cni/net.d \
+             --mount volume=etc-cni,target=/etc/cni/net.d
+
+    * Mount the directory where CNI plugins are installed:
+
+        ::
+
+             --volume cni-bin,kind=host,source=/opt/cni/bin \
+             --mount volume=cni-bin,target=/opt/cni/bin
+
+    * `Mount the BPF filesystem <http://docs.cilium.io/en/latest/admin/#mounting-the-bpf-fs>`_ so that the information stored there persists across Cilium restarts:
+
+        ::
+             
+             ExecStartPre=/bin/mount bpffs /sys/fs/bpf -t bpf
+
+    * Also ensure that you specify that the network plugin is CNI: 
+
+        ::  
+
+             --network-plugin=cni
+
+    * Specify the CNI directory to correspond to the mount you provided earlier where the CNI configuration is located:
+    
+        ::
+
+             --cni-conf-dir=/etc/cni/net.d
+
+`/etc/systemd/system/kubelet.service`
+:: 
+   [Service]
+   Environment=KUBELET_IMAGE_TAG=v'"${KUBE_VERSION}"'_coreos.0
+   Environment="RKT_RUN_ARGS=--uuid-file-save=/var/run/kubelet-pod.uuid \
+     --volume var-log,kind=host,source=/var/log \
+     --mount volume=var-log,target=/var/log \
+     --volume dns,kind=host,source=/etc/resolv.conf \
+     --mount volume=dns,target=/etc/resolv.conf \
+     --volume cni-bin,kind=host,source=/opt/cni/bin \
+     --mount volume=cni-bin,target=/opt/cni/bin \
+     --volume etc-cni,kind=host,source=/etc/cni/net.d \
+     --mount volume=etc-cni,target=/etc/cni/net.d"
+   ExecStartPre=/usr/bin/mkdir -p /etc/cni/net.d
+   ExecStartPre=/bin/mount bpffs /sys/fs/bpf -t bpf
```

Comment:
Can you add a comment why is this here? And / or write a comment before this kubelet.service so users know this is needed for cilium?

Also this is more correct:
```
ExecStartPre=/bin/bash -c ' \\
 if [[ \$(/bin/mount | /bin/grep /sys/fs/bpf -c) -eq 0 ]]; then \\
   /bin/mount bpffs /sys/fs/bpf -t bpf; \\
 fi'
```

otherwise the user might end up with multiple mounts of the bpffs whenever does `sudo service kubelet restart`